src/instructlab/sdg/generate_data.py

```diff
@@ -164,9 +165,20 @@ def _gen_test_data(
             outfile.write("\n")
 
 
+def _check_pipeline_dir(pipeline):
+    for file in ["knowledge.yaml", "freeform_skills.yaml", "grounded_skills.yaml"]:
+        if not os.path.exists(os.path.join(pipeline, file)):
+            raise GenerateException(
+                f"Error: pipeline directory ({pipeline}) does not contain {file}."
+            )
+
+
 def _sdg_init(pipeline, client, model_family, model_id, num_instructions_to_generate):
     pipeline_pkg = None
-    if pipeline == "full":
+    if os.path.exists(os.path.join(CUSTOM_PIPELINES_DATA_DIR, pipeline)):
+        pipeline = os.path.join(CUSTOM_PIPELINES_DATA_DIR, pipeline)
+        _check_pipeline_dir(pipeline)
+    elif pipeline == "full":
         pipeline_pkg = FULL_PIPELINES_PACKAGE
     elif pipeline == "simple":
         pipeline_pkg = SIMPLE_PIPELINES_PACKAGE
```

Comment:
It's tempting to say that these pre-defined ones should go into `/usr/share/` too ... except more like a `DATA_DIR` specific to a the virtual environment the package is installed into? Consider it if you find something obvious, but otherwise let's leave it as it is